staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -480,8 +492,19 @@ func (r *Request) Watch() (watch.Interface, error) {
 	if err != nil {
 		return nil, err
 	}
+	// If request r's context is nil or has no deadline, use the requestTimeout
 	if r.ctx != nil {
-		req = req.WithContext(r.ctx)
+		if _, ok := r.ctx.Deadline(); ok == true {
+			req = req.WithContext(r.ctx)
+		} else {
+			c, cancel := context.WithTimeout(r.ctx, r.requestTimeout)
+			req = req.WithContext(c)
+			defer cancel()
```

Comment:
This function returns a watch interface so the request is not finished when the function is returned. This cancel should be somehow part of the Watch.Stop but also we need to make sure it will run if we did not return the interface (e.g. failed to request).

Reply:
@mbohlool Done. Also updated Stream(), which returns a cancel function as well.